torch/quantization/_correct_bias.py

```diff
@@ -0,0 +1,292 @@
+            reassign[name] = ShadowModule(float_mod, mod)
+
+    for key, value in reassign.items():
+        quantized_model._modules[key] = value
+
+def remove_hooks(model, white_list=_supported_modules):
+    ''' Removes all the hooks from the whitelisted submodules
+    '''
+    for name, module in model.named_modules():
+        if type(module) in white_list:
+            if hasattr(module, 'activation_post_process'):
+                for key in module._forward_hooks:
+                    del module._forward_hooks[key]
+                del module.activation_post_process
+            if hasattr(module, 'activation_pre_process'):
+                for key in module._forward_pre_hooks:
+                    del module._forward_pre_hooks[key]
+                del module.activation_pre_process
+
+def remove_shadow(model):
+    ''' Goes through the submodules of a quantized model and replaces every
+    shadowModule with just the quantized submodule it contains
+    '''
+    reassign = {}
+    # logging occurences of shadowModules
+    for name, module in model.named_modules():
+        if isinstance(module, ShadowModule):
+            reassign[name] = module.quantized_module
+    # removing occurences of shadowModules
+    for name, module in reassign.items():
+        name_split = name.split('.')
+        get_parent_module(model, name)._modules[name_split[-1]] = reassign[name]
+
+def sequential_bias_correction(float_model, quantized_model, img_data, white_list=_supported_modules):
+    ''' Applies bias correction on the whitelisted submodules of the quantized model
+    After adding shadow modules and MeanLoggers to the quantized model the img_data will get reshaped
+    into one large batch and then this batch will be ran through the quantized model
+
+    at every whitelisted submodule during this execution, bias correction will be applied
+    on the quantized submodule within the shadowModule, before moving to the next submodule
+    '''
+    # adding hooks
+    # setting up qconfigs to add post hook
+    qconfig_debug = torch.quantization.QConfig(activation=MeanLogger, weight=None)
+    float_model.qconfig = qconfig_debug
+    quantized_model.qconfig = qconfig_debug
+    # calling prepare with prehook param, and throwing the whitelist to make sure supported modules
+    # get the qconfig and thus post hook
+    torch.quantization.prepare(float_model, inplace=True, white_list=white_list, prehook=MeanLogger)
+    torch.quantization.prepare(quantized_model, inplace=True, white_list=white_list,
+                                observer_non_leaf_module_list=[nnq.Linear], prehook=MeanLogger)
+
+    add_shadow(float_model, quantized_model, white_list=white_list)
+
+    # reshaping img_data into one large batch
+    stack = None
+    for data in img_data:
+        if stack is None:
+            stack = data[0]
+        else:
+            stack = torch.cat((stack, data[0]))
+
+    # bias correction is happening under the hood here
```

Comment:
bias correction should be an explicit step, it should not happen under the hood

Reply:
i see why you are calling this sequential now. this is not what we expect for sequential, please see Raghu's comment for the steps for sequential.